src/Options.js

```diff
@@ -85,7 +85,7 @@ class Options {
      * Returns a new options object with the same properties as "this".  Properties
      * from "extension" will be copied to the new options object.
      */
-    extend(extension: OptionsData): Options {
+    extend(extension: $Shape<OptionsData>): Options {
```

Comment:
Can you remove `$Shape` here as well?  I think it may be hiding some null checking we should be doing.

Reply:
I think this `$Shape` here is necessary since you should able to do stuff like `newOptions = options.extend({color: 'white'})`, which doesn't set the required `style` and `maxSize` properties in `OptionData`. You still wouldn't be able to do stuff like `options.extend({style: null})` (see [Flow docs](http://sitr.us/2015/05/31/advanced-features-in-flow.html#shapet)).